src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala

```diff
@@ -0,0 +1,169 @@
+    * @tparam F
+    *   The type of the effect
+    * @tparam S
+    *   The type of the Stream to be used for the streaming methods.
+    * @return
+    *   A new `ProcessMonitorClient`
+    */
+  def createProcessMonitorClient[F[_], S](clientConfig: ClientConfig[F, S])(implicit
+      me: MonadError[F, Throwable],
+      sync: Sync[F]
+  ): ProcessMonitorClient[F] = new ProcessMonitorClient[F] {
+    private val apiBaseUrl: String = clientConfig.apiBaseUrl
+    private val client: Client[F, S] = Client(clientConfig)
+
+    implicit val monitorsPayloadRW: ReadWriter[Monitors] = macroRW[Monitors]
+
+    implicit val pagingPayloadRW: ReadWriter[Paging] = macroRW[Paging]
+
+    implicit val valuePayloadRW: ReadWriter[Value] = macroRW[Value]
+
+    implicit val monitorsResponsePayloadRW: ReadWriter[MonitorsResponse] = macroRW[MonitorsResponse]
+
+    import client._
+
+    override def getMonitors(getMonitorsRequest: GetMonitorsRequest): F[Seq[Monitors]] = {
+      val getMonitorsRequestAsMap: Map[String, IterableOnce[String]] =
+        getMonitorsRequest.productElementNames
+          .zip(
+            getMonitorsRequest.productIterator.map(_.asInstanceOf[IterableOnce[String]])
+          )
+          .toMap
+      val relevantQueryParamsAsString: Map[String, String] = getMonitorsRequestAsMap.collect {
+        case (name, value) if value.iterator.nonEmpty => (name, value.iterator.mkString(","))
+      }
+
+      val getMonitorsUrl = uri"$apiBaseUrl/api/processmonitor/monitors?$relevantQueryParamsAsString"
+
+      for {
+        _ <-
+          if (!relevantQueryParamsAsString("name").startsWith("opex"))
+            me.raiseError(
+              PreservicaClientException(
+                "The monitor name must start with 'opex'"
+              )
+            )
+          else me.unit
+        token <- getAuthenticationToken
+        getMonitorsResponse <- sendJsonApiRequest[MonitorsResponse](
+          getMonitorsUrl.toString,
+          token,
+          Method.GET,
+          responseSchema = asJson[MonitorsResponse]
+        )
+      } yield getMonitorsResponse.value.monitors
+    }
+  }
+
+  sealed trait MonitorsStatus {
+    override def toString: String = getClass.getSimpleName.dropRight(1)
+  }
+
+  /** A Monitors request parameter
+    * @param key
+    *   The parameter key
+    * @param value
+    *   The parameter value
+    */
+  case class Parameter(key: String, value: String)
+
+  /** A Monitors' request
+    * @param status
+    *   An optional monitor status.
+    * @param name
+    *   An optional monitor name.
+    * @param category
+    *   A list of categories. This can be empty
+    */
+  case class GetMonitorsRequest(
+      status: List[MonitorsStatus] = Nil,
+      name: Option[String] = None,
+      category: List[MonitorCategory] = Nil
+  ) {
+    val subcategory: List[String] = Nil
+  }
+
+  case class MonitorsError(code: String, description: String, status: String, file: String)
```

Comment:
This case class isn't used anywhere. Do we need it?

Reply:
Ah, no; this was a remnant of some other code.